When insecure=true, the provider registry.terraform.io/smutel/netbox failed

<!--- BEFORE CREATING A NEW ISSUE, ENSURE THAT THE PROVIDER VERSION THAT YOUR ARE USING IS -->
<!--- COMPATIBLE WITH YOUR VERSION OF NETBOX. PLEASE CHECK THE COMPATIBILITY MATRIX HERE -->
<!--- https://github.com/smutel/terraform-provider-netbox#compatibility-with-netbox -->

<!--- Verify first that your issue is not already reported on GitHub -->
<!--- Ensure that the latest release is affected by this bug -->
<!--- Complete most of sections below as described -->

## Summary
panic: runtime error: invalid memory address or nil pointer dereference when insecure=true

## Version
7.1.0
### Netbox version
<!--- Enter below the version of netbox -->
```
3.4.10
```

### Terraform version
<!--- Enter below the result of "terraform -v" -->
```
1.3.6
```

### Provider version
<!--- Enter below the version of terraform-provider-netbox -->
```
v7.1.0
```

## Issue details

### Affected Data(s) / Resource(s)
<!--- Give the name of the data(s) or resource(s) affected by this bug -->

### Terraform Configuration Files
<!-- Copy-paste your Terraform configurations below -->
<!-- For large Terraform configs, please give a link to a https://gist.github.com -->
```
provider "netbox" {
 url  = "..."
 token   = ..."
 scheme = "https"
 insecure = true
}
```

### Terraform Output
<!-- Copy-paste the terraform output (only the error) -->
```
Error: Plugin did not respond
│ 
│   with provider["registry.terraform.io/smutel/netbox"],
│   on main.tf line 12, in provider "netbox":
│   12: provider "netbox" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ConfigureProvider call. The plugin logs may contain more details.
```

## Behaviors

### Actual Behavior
When insecure=true, the provider registry.terraform.io/smutel/netbox failed

### Expected Behavior
If you change the line :
t.Transport.(*http.Transport).TLSClientConfig.InsecureSkipVerify = insecure
by
t.Transport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: insecure}
in provider.go
The parameter insecure=true runs correctly 

### Steps to Reproduce
<!-- Please list the steps required to reproduce the issue, for example:-->
1. Set `insecure=true` in the provider's parameters 
2. `terraform plan`

--- a/netbox/provider.go
+++ b/netbox/provider.go
@@ -2,6 +2,7 @@ package netbox
 
 import (
 	"context"
+	"crypto/tls"
 	"fmt"
 	"net/http"
 
@@ -227,7 +228,7 @@ func configureProvider(ctx context.Context, d *schema.ResourceData) (interface{}
 
 	t := runtimeclient.New(url, basepath, defaultScheme)
 	if insecure {
-		t.Transport.(*http.Transport).TLSClientConfig.InsecureSkipVerify = insecure
+		t.Transport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: insecure} // #nosec G402
 	}
 	t.DefaultAuthentication = runtimeclient.APIKeyAuth(authHeaderName, "header",
 		fmt.Sprintf(authHeaderFormat, token))

